ArrayIndexOutOfBoundsException when playlist is empty

- [x] I have verified there are no duplicate active or recent bugs, questions, or requests

- PlaylistCore version: `1.1.1` (probably lower version too)
- Device OS version: `6.0`  (probably lower version too)
- Devide Manufacturer: `Huawei` (+emulator)
- Device Name: `Nexus 6P` (+emulator)
 
###### Reproduction Steps
1. Initialise the playlist manager
2. Kill app

###### Expected Result

No crash expected

###### Actual Result

```
java.lang.ArrayIndexOutOfBoundsException: length=0; index=-1
    at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3314)
    at android.app.ActivityThread.-wrap21(ActivityThread.java)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1565)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:154)
    at android.app.ActivityThread.main(ActivityThread.java:6077)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755)
 Caused by: java.lang.ArrayIndexOutOfBoundsException: length=0; index=-1
    at java.util.ArrayList.get(ArrayList.java:413)
    at com.devbrackets.android.playlistcore.manager.ListPlaylistManager.getItem(ListPlaylistManager.java:81)
    at com.devbrackets.android.playlistcore.manager.BasePlaylistManager.findNextAllowedPosition(BasePlaylistManager.java:707)
    at com.devbrackets.android.playlistcore.manager.BasePlaylistManager.setCurrentPosition(BasePlaylistManager.java:412)
    at com.devbrackets.android.playlistcore.service.PlaylistServiceCore.relaxResources(PlaylistServiceCore.java:1009)
    at com.devbrackets.android.playlistcore.service.BasePlaylistService.relaxResources(BasePlaylistService.java:223)
    at com.devbrackets.android.playlistcore.service.PlaylistServiceCore.onDestroy(PlaylistServiceCore.java:285)
    at com.devbrackets.android.playlistcore.service.BasePlaylistService.onDestroy(BasePlaylistService.java:168)
    at com.devbrackets.android.playlistcore.service.PlaylistServiceCore.onTaskRemoved(PlaylistServiceCore.java:329)
    at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3299)
    at android.app.ActivityThread.-wrap21(ActivityThread.java) 
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1565) 
    at android.os.Handler.dispatchMessage(Handler.java:102) 
    at android.os.Looper.loop(Looper.java:154) 
    at android.app.ActivityThread.main(ActivityThread.java:6077) 
    at java.lang.reflect.Method.invoke(Native Method) 
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865) 
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755) 
```

--- a/library/src/main/java/com/devbrackets/android/playlistcore/manager/BasePlaylistManager.java
+++ b/library/src/main/java/com/devbrackets/android/playlistcore/manager/BasePlaylistManager.java
@@ -704,7 +704,7 @@ public abstract class BasePlaylistManager<I extends IPlaylistItem> implements Pl
             return getItemCount();
         }
 
-        while (position < getItemCount() && !isAllowedType(getItem(position))) {
+        while (position < getItemCount() && position >= 0 && !isAllowedType(getItem(position))) {
             position++;
         }
 

